### PR TITLE
[INFINITY-1586] User must type name of service to confirm uninstall

### PR DESCRIPTION
**Pull request by benclarkwood**
WIP. Creating to run integration tests.

### CODE REVIEW

File: cli/dcoscli/util.py [hunk 1]
```diff
@@ -54,3 +54,24 @@ def confirm(prompt, yes):
...
+    """
+    :param prompt: message to display to the terminal
+    :type prompt: str
+    :param expected_text: The text to compare user input to.
+    :type expected_text: str
+    :returns: True if the user replies with the expected text; False otherwise
+    :rtype: bool
+    """
+
+    while True:
```
Comment (jsrodman) [inline cli/dcoscli/util.py]: What does this loop do?

File: cli/dcoscli/util.py [hunk 2]
```diff
@@ -54,3 +54,24 @@ def confirm(prompt, yes):
...
+        sys.stdout.write('{}: '.format(prompt))
+        sys.stdout.flush()
+        response = sys.stdin.readline().strip()
+        if response == confirmation_text:
+            return True
+        else:
+            msg = "Expected '{}'. You supplied '{}'.".format(confirmation_text, response)
```
Comment (jsrodman) [inline cli/dcoscli/util.py]: Suggestion: confirmText's  caller should probably pass outcome of failure in as text.  (maybe rename confirm_text)

File: cli/tests/integrations/test_package.py [hunk 3]
```diff
@@ -882,6 +882,7 @@ def _uninstall_helloworld(
         stderr=b'',
         returncode=0,
         uninstalled=b'Uninstalled package [helloworld] version [0.1.0]\n'):
+    args = args.append("--confirm_uninstall")
```
Comment (jsrodman) [inline cli/tests/integrations/test_package.py]: mentioned in chat, but this wants to be 
```python
args.append("--confirm_uninstall")
```

**Comment by jsrodman** [comment]
I guess the build failure is unrelated:

```
[test-install-script-1-build-docker-image] $ /usr/bin/env bash /tmp/hudson7420525513499356649.sh
Building docker image...
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /jenkins/workspace/DCOS-CLI/test-install-script-1-build-docker-image/ccm/Dockerfile: no such file or directory
Build step 'Execute shell' marked build as failure
```

**Comment by benclarkwood** [comment]
See Karl's comments on Jesus' PR. New build that isn't related.

**Comment by benclarkwood** [comment]
@tamarrow Ready for review

File: cli/dcoscli/data/help/package.txt [hunk 4]
```diff
@@ -21,7 +21,7 @@ Usage:
     dcos package repo remove <repo-names>...
     dcos package search [<query> --json]
     dcos package uninstall <package-name>
-                           [--cli | [--app --app-id=<app-id> --all]]
+                           [--cli | [--app --app-id=<app-id> --all --yes]]
```
Comment (benclarkwood) [inline cli/dcoscli/data/help/package.txt]: Correct, the user will only need to confirm something when uninstalling an app.

They will specifically acknowledge _what_ app they are uninstalling

File: cli/dcoscli/package/main.py [hunk 5]
```diff
@@ -488,3 +493,36 @@ def _uninstall(package_name, remove_all, app_id, cli, app):
         return 1
 
     return 0
+
+def __confirm_uninstall(package_name, remove_all, app_id):
```
Comment (jsrodman) [inline cli/dcoscli/package/main.py]: Double underscore is weird.  This is a pythonism for "mangle this symbol to do fake data hiding", which probably shouldn't have been added to the language in the first place.
Single underscore is a convention for "don't show this object to people importing this module the normal way".

**Comment by jsrodman** [comment]
FWIW, I can't make sense of the binary build problem.  A  lot of innocuous INFO messages, then a failed message with no hints.  I'm suspicious this problem may be unrelated to this change?

File: cli/dcoscli/package/main.py [hunk 6]
```diff
@@ -476,10 +476,15 @@ def _uninstall(package_name, remove_all, app_id, cli, app):
     :type remove_all: boolean
     :param app_id: App ID of the package instance to uninstall
     :type app_id: str
+    :param skip_confirmation: Skip confirmation for uninstall
+    :type skip_confirmation: boolean
     :returns: Process status
     :rtype: int
     """
 
+    if not skip_confirmation and not cli and not _confirm_uninstall(package_name, remove_all, app_id):
```
Comment (tamarrow) [inline cli/dcoscli/package/main.py]: from docopt you cant have `--yes` and `--cli` so you dont need `not cli` here.

File: cli/dcoscli/package/main.py [hunk 7]
```diff
@@ -476,10 +476,15 @@ def _uninstall(package_name, remove_all, app_id, cli, app):
...
+        return 0
```
Comment (tamarrow) [inline cli/dcoscli/package/main.py]: we should return 1 here, since there was an error

File: cli/tests/data/helloworld-v3-stub.json [hunk 8]
```diff
@@ -26,7 +26,7 @@
           "containerPort": 80
         }
       ],
-      "image": "mesosphere/universe-server:20170331T221320Z-cli-test-v3-helloworld-c3e49742c9",
+      "image": "mesosphere/universe-server:20170518T235510Z-cli-test-v3-helloworld-ca5778937b",
```
Comment (tamarrow) [inline cli/tests/data/helloworld-v3-stub.json]: Looks like a problem with your rebase, this is pulled from #998

**Comment by tamarrow** [comment]
can we add a unit test for confirm_uninstall please? The rest of the comments are very minor. The build has been re-run but I assume the problem mentioned above is syntax errors. At the bottom of the log it says which tests passed and failed. You can run syntax locally by running `tox -e py35-syntax` in the top directory and in the cli directory.

**Comment by benclarkwood** [comment]
@tamarrow Unit test added (as well as syntax fixes), and properly rebased.

Will see how the tests do.

**Pull request merged**